### PR TITLE
[Breaking] Only support node 10.* && >= 12.*

### LINKED ISSUE #9: [Updates] Use async/await directly
Now that Node 8 is nearly EOL, we can safely re-write this code to use async/await. as Promises in node 10 have `finally`. This will provide improved debugging and stacktraces, and drop some over our heavier dependencies.

### PR DESCRIPTION
**Pull request by stefanpenner**
[Fixes #9]

* enables use of async/await for improved debugging and stack traces
* drops rsvp dependency

### CODE REVIEW

**Comment by stefanpenner** [comment]
@joliss also, I would gladly help maintain + reduce the buss factor for this node_module. If you are ok with that, feel free to add me as a maintainer.

File: test.js [hunk 1]
```diff
@@ -1,16 +1,15 @@
-var test = require('tape')
-var RSVP = require('rsvp')
-var mapSeries = require('./index')
-
+'use stricdt';
```
Comment (rwjblue) [inline test.js]: small typo here
```suggestion
'use strict';
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,12 @@
 # master
 
+# 1.0.0
+
+* [BREAKING] Drop support for any node but `10.* || >= 12.*`
+* Use async/await directly
+* Drop RSVP
+* Upgrade dependencies
```
Comment (rwjblue) [inline CHANGELOG.md]:
```suggestion
* Upgrade testing dependencies
```

**Comment by stefanpenner** [comment]
@rwjblue thanks!

**Comment by joliss** [comment]
Thanks, and sorry for the long delay! I've released this as v0.3.0 and added you both as owners of this package!

**Comment by stefanpenner** [comment]
@joliss no worries, thanks for merging. I am excited native promises and async/await are ready for primetime!

**Pull request merged**